definitions/tools/mutect.cwl

```diff
@@ -2,75 +2,62 @@
+
 requirements:
     - class: ResourceRequirement
       ramMin: 20000
       tmpdirMin: 100000
     - class: DockerRequirement
-      dockerPull: "mgibio/cle:v1.3.1"
+      dockerPull: "broadinstitute/gatk:4.1.2.0"
+    - class: InitialWorkDirRequirement
+      listing:
+      - entryname: 'Mutect2.sh'
+        entry: |
+            set -o pipefail
+            set -o errexit
+            
+            TUMOR=`(samtools view -H $3 | grep "^@RG" | sed "s/.*SM:\([^\t]*\).*/\1/g" | head -n 1)` #Extracting the sample name from the TUMOR bam.
+            NORMAL=`(samtools view -H $4 | grep "^@RG" | sed "s/.*SM:\([^\t]*\).*/\1/g" | head -n 1)` #Extracting the sample name from the NORMAL bam.
+            /gatk/gatk Mutect2 -O $1 -R $2 -I $3 -tumor $TUMOR -I $4 -normal $NORMAL -L $5 #Running Mutect2.
+            gunzip mutect.vcf.gz #Unzipping the Mutect2 output vcf file to alter the header names.
+            grep "^##" mutect.vcf >mutect.final.vcf #Extracting all the lines above #CHROM in the vcf to a new output vcf.
+            grep "^#CHROM" mutect.vcf | sed "s/\t$TUMOR\t/\tTUMOR\t/g; s/\t$NORMAL/\tNORMAL/g" >> mutect.final.vcf #Concatenating the #CHROM line with substituted sample headers to the output vcf.
+            grep -v "^#" mutect.vcf >> mutect.final.vcf #Concatenating the sites to the output vcf.
+            bgzip mutect.final.vcf #Bzipping the final output vcf.
+            tabix mutect.final.vcf.gz #Indexing the bzipped output vcf.
+            mv mutect.vcf.gz.stats mutect.final.vcf.gz.stats #Renaming the .stats file to match the name conversion of the final output vcf.
+            /gatk/gatk FilterMutectCalls -R $2 -V mutect.final.vcf.gz -O mutect.final.filtered.vcf.gz #Running FilterMutectCalls on the output vcf.
+
 arguments:
-    ["-o", { valueFrom: $(runtime.outdir)/mutect.vcf.gz }]
+    - position: 1
+      valueFrom: mutect.vcf.gz
+
 inputs:
     reference:
         type: string
         inputBinding:
-            prefix: "-R"
-            position: 1
+            position: 2
     tumor_bam:
         type: File
         inputBinding:
-            prefix: "-I:tumor"
-            position: 2
+            position: 3
         secondaryFiles: [.bai]
     normal_bam:
         type: File?
         inputBinding:
-            prefix: "-I:normal"
-            position: 3
+            position: 4
         secondaryFiles: [.bai]
     interval_list:
         type: File
         inputBinding:
-            prefix: "-L"
-            position: 4
-    dbsnp_vcf:
-        type: File?
-        inputBinding:
-            prefix: "--dbsnp"
             position: 5
-        secondaryFiles: [.tbi]
-    cosmic_vcf:
-        type: File?
-        inputBinding:
-            prefix: "--cosmic"
-            position: 6
-        secondaryFiles: [.tbi]
-    artifact_detection_mode:
-        type: boolean?
-        inputBinding:
-            prefix: "--artifact_detection_mode"
-            position: 7 
-    panel_of_normals_vcf:
```

Comment:
It is probably not a bad idea - as outlined here, it will still accept a PON, although I think we're going to have to generate a new one:
 https://gatkforums.broadinstitute.org/gatk/discussion/10911/differences-between-gatk3-mutect2-and-gatk4-mutect2

> Mutect2 ignores sites present in the Panel of Normals (PoN) as well as sites that correspond to high fraction variants in the normal. By doing so, the tool avoids spending time in steps such as graph assembly and pairHMM alignments that cost compute. However, there is an option to force the tool to run the full process on sites that are in the PoN (--genotype-pon-sites), which can be useful in comparing results to older MuTect versions.
> 
> To create the PoN, call on each normal sample using Mutect2's tumor-only mode and then use GATK4's CreateSomaticPanelOfNormals, a tool new to GATK4. This contrasts with the GATK3 workflow, which uses an artifact calling mode in MuTect2 and CombineVariants for PoN creation. In GATK4, omitting to filter with FilterMutectCalls achieves the same result.

Reply:
Do we have a manifest of the samples used to create the PON?  Perhaps we should we round up some samples and generate a new one using some things sequenced on the NovaSeq?